### ISSUE TITLE
use LinearAlgebra's new generic in-place 5-arg mul! wherever possible (requires Julia 1.3)

### ISSUE THREAD

**Issue by chriscoey** [JuliaMath/IterativeSolvers.jl#254]
see https://github.com/JuliaLang/julia/pull/29634

this will fix #206, #245 and is relevant to #251, #248

**Comment by davidschlegel** [comment]
I think this will also fix [#295 ](https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/issues/295).

**Comment by ranocha** [comment]
I think #295 will also require updates of lines like https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/ae01dfe228138d0db9164eb343252150f6c0bfc6/src/bicgstabl.jl#L37-L40

**Comment by davidschlegel** [comment]
Yes, and probably at some other lines. I am not familiar with the code, so I think I cannot really help here, but I would love to have this fixed. The cg function already has the functionality to work with CUDA types, and so I think it should be possible to do it here are well.
